src/Avalonia.X11/X11Screens.cs

```diff
@@ -97,24 +130,13 @@ public unsafe X11Screen[] Screens
                         var namePtr = XGetAtomName(_x11.Display, mon.Name);
                         var name = Marshal.PtrToStringAnsi(namePtr);
                         XFree(namePtr);
-
+                        var bounds = new PixelRect(mon.X, mon.Y, mon.Width, mon.Height);
                         var density = 1d;
-                        if (_settings.NamedScaleFactors?.TryGetValue(name, out density) != true)
-                        {
-                            if (mon.MWidth == 0)
-                                density = 1;
-                            else
-                                density = X11Screen.GuessPixelDensity(mon.Width, mon.MWidth);
-                        }
-
+                        var pSize = GetPhysicalMonitorSizeFromEDID(mon.Outputs);
```

Comment:
I guess we should also check `NOutputs` to be `>= 1`. BTW, any idea why it can be more than 1?

Reply:
This can be the case if the user creates an monitor with multiple outputs attache by running something like this:
```
$ xrandr --setmonitor DP-2-1 auto DP-2,HDMI-1
$ xrandr --listmonitors
Monitors: 1
 0: DP-2-1 5120/0x1440/1+0+0  DP-2 HDMI-1
```
So detecting the smallest scaling factor among the attached outputs should be safest way to handle this case.